src/material/input/testing/native-select-harness.ts

```diff
@@ -0,0 +1,98 @@
+  static hostSelector = 'select[matNativeControl]';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatNativeSelectHarness` that meets
+   * certain criteria.
+   * @param options Options for filtering which select instances are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: NativeSelectHarnessFilters = {}): HarnessPredicate<MatNativeSelectHarness> {
+    return new HarnessPredicate(MatNativeSelectHarness, options);
+  }
+
+  /** Gets a boolean promise indicating if the select is disabled. */
+  async isDisabled(): Promise<boolean> {
+    return (await this.host()).getProperty('disabled');
+  }
+
+  /** Gets a boolean promise indicating if the select is required. */
+  async isRequired(): Promise<boolean> {
+    return (await this.host()).getProperty('required');
+  }
+
+  /** Gets a boolean promise indicating if the select is in multi-selection mode. */
+  async isMultiple(): Promise<boolean> {
+    return (await this.host()).getProperty('multiple');
+  }
+
+  /** Gets the name of the select. */
+  async getName(): Promise<string> {
+    // The "name" property of the native select is never undefined.
+    return (await (await this.host()).getProperty('name'))!;
+  }
+
+  /** Gets the id of the select. */
+  async getId(): Promise<string> {
+    // We're guaranteed to have an id, because the `matNativeControl` always assigns one.
+    return (await (await this.host()).getProperty('id'))!;
+  }
+
+  /** Focuses the select and returns a void promise that indicates when the action is complete. */
+  async focus(): Promise<void> {
+    return (await this.host()).focus();
+  }
+
+  /** Blurs the select and returns a void promise that indicates when the action is complete. */
+  async blur(): Promise<void> {
+    return (await this.host()).blur();
+  }
+
+  /** Whether the select is focused. */
+  async isFocused(): Promise<boolean> {
+    return (await this.host()).isFocused();
+  }
+
+  /** Gets the options inside the select panel. */
+  async getOptions(filter: NativeOptionHarnessFilters = {}):
+    Promise<MatNativeOptionHarness[]> {
+    return this.locatorForAll(MatNativeOptionHarness.with(filter))();
+  }
+
+  /**
+   * Selects the options that match the passed-in filter. If the select is in multi-selection
+   * mode all options will be clicked, otherwise the harness will pick the first matching option.
+   */
+  async selectOptions(filter: NativeOptionHarnessFilters = {}): Promise<void> {
+    const [isMultiple, options] = await Promise.all([this.isMultiple(), this.getOptions(filter)]);
+
+    if (options.length === 0) {
+      throw Error('Select does not have options matching the specified filter');
+    }
+
+    const [host, optionIndexes] = await Promise.all([
+      this.host(),
+      Promise.all(options.slice(0, isMultiple ? undefined : 1).map(option => option.getIndex()))
+    ]);
+
+    await host.selectOptions!(...optionIndexes);
```

Comment:
Should we throw some more friendly error if this fails? like "The current HarnessEnvironment does not yet support selecting options on a native select" or something